src/Functions/tupleElement.cpp

```diff
@@ -103,7 +119,11 @@ class FunctionTupleElement : public IFunction
         if (!tuple_type_concrete || !tuple_col_concrete)
             throw Exception("First argument for function " + getName() + " must be tuple or array of tuple.", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 
-        size_t index = getElementNum(arguments[1].column, *tuple_type_concrete);
+        size_t index = 0;
+        if (!getElementNum(arguments[1].column, *tuple_type_concrete, index, arguments.size()))
+        {
+            return arguments[2].column;
+        }
```

Comment:
In case of `Array(Tuple))` it may work incorrectly. In that case we should return array with same sizes as other elements have, but now arbitrary value of any type may be returned.

 So, in case of `Array(Tuple))` we have to check that default value has `Array` type with the same number of dimension as the original column and check that it has the same offsets in all rows as the original column.